zenoh/tests/acl.rs

```diff
@@ -1748,3 +1738,200 @@ async fn test_liveliness_deny_allow_query(port: u16) {
+                {
+                    "rules": ["deny query egress"],
+                    "subjects": ["all"],
+                }
+            ],
+          }"#,
+        )
+        .unwrap();
+
+    let _router = ztimeout!(zenoh::open(config_router)).unwrap();
+    let (session1, session2) = get_client_sessions(27502).await;
+    tokio::time::sleep(SLEEP).await;
+
+    let _qbl = session1.declare_queryable("test/egress").await.unwrap();
+    tokio::time::sleep(SLEEP).await;
+    let replies = session2.get("test/egress").await.unwrap();
+
+    assert!(replies.recv_async().await.is_err());
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+async fn test_acl_liveliness_query_ingress_deny() {
+    zenoh::init_log_from_env_or("error");
+    let mut config_router = get_basic_router_config(27503).await;
+    config_router
+        .insert_json5(
+            "access_control",
+            r#"{
+            "enabled": true,
+            "default_permission": "allow",
+            "rules": [
+                {
+                    "id": "deny query ingress",
+                    "permission": "deny",
+                    "messages": ["query", "liveliness_token", "liveliness_query"],
+                    "flows": ["ingress"],
+                    "key_exprs": ["**"],
+                }
+            ],
+            "subjects": [
+                { "id": "all" }
+            ],
+            "policies": [
+                {
+                    "rules": ["deny query ingress"],
+                    "subjects": ["all"],
+                }
+            ],
+          }"#,
+        )
+        .unwrap();
+
+    let _router = ztimeout!(zenoh::open(config_router)).unwrap();
+    tokio::time::sleep(SLEEP).await;
+    let (session1, session2) = get_client_sessions(27503).await;
+    tokio::time::sleep(SLEEP).await;
+
+    let _token = session1
+        .liveliness()
+        .declare_token("test/token/ingress")
+        .await
+        .unwrap();
+    tokio::time::sleep(SLEEP).await;
+
+    let replies = session2
+        .liveliness()
+        .get("test/token/ingress")
+        .timeout(Duration::from_secs(10))
+        .await
+        .unwrap();
+
+    assert!(replies.recv_timeout(Duration::from_secs(1)).is_err());
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+async fn test_acl_liveliness_query_egress_deny() {
+    zenoh::init_log_from_env_or("error");
+    let config_router = get_basic_router_config(27504).await;
+    let mut config_client1 = get_basic_client_config(27504).await;
+    let config_client2 = get_basic_client_config(27504).await;
+    config_client1
+        .insert_json5(
+            "access_control",
+            r#"{
+            "enabled": true,
+            "default_permission": "allow",
+            "rules": [
+                {
+                    "id": "deny query egress",
+                    "permission": "deny",
+                    "messages": ["query", "liveliness_token", "liveliness_query"],
```

Comment:
`"query"` and `"liveliness_token"` messages are not related to this test
```suggestion
                    "messages": ["liveliness_query"],
```

Reply:
fixed